aa_engine/aas_retrieve_inputs_part2.m

```diff
@@ -59,20 +59,23 @@
                             aas_log(aap,false,sprintf(' retrieve remote stream %s from %s:%s to %s',streamfiles(depind).streamname,streamfiles(depind).inputstream.host,logsafe_path_src,logsafe_path_dest),aap.gui_controls.colours.inputstreams);
 
                             oldpth='';
-
                             % rsync in chunks. It will then compress.
                             chunksize=64;
                             transfernow=false;
                             numtotransfer=0;
                             inps='';
+                            use_remotesymlinks = 0;
                             for ind=1:length(streamfiles(depind).fns)
                                 % Copy file
                                 [pth,nme,ext]=fileparts(streamfiles(depind).fns_dest_full{ind});
                                 newpth=pth;
+                                if (~streamfiles(depind).ismodified & aap.options.remotesymlinks == 1)
```

Comment:
I think you could move these three lines (72-74) before the for loop (i.e., after line 67).

Reply:
Please see your comment and our discussion here: https://github.com/automaticanalysis/automaticanalysis/issues/361#issuecomment-1322567375

That part needs to be in the loop, because otherwise how will it know that it needs to copy which files?

Reply:
The `ismodified` flag is for the whole stream and not for individual files. Your solution as is, is working, of course, but not efficient because the conditions will not change across the files within the sam stream.